contracts/hackatom/src/contract.rs

```diff
@@ -219,6 +223,44 @@ fn do_panic() -> StdResult<HandleResponse> {
     panic!("This page intentionally faulted");
 }
 
+fn do_user_errors_in_api_calls<A: Api>(api: &A) -> StdResult<HandleResponse> {
+    // Canonicalize
+
+    let empty = HumanAddr::from("");
+    match api.canonical_address(&empty).unwrap_err() {
+        StdError::GenericErr { .. } => {}
+        err => panic!("Unexpected error: {:?}", err),
```

Comment:
If you panic, there is no useful error message outside of the vm.

Can you change these all to `Err(StdError::generic(...))` so the message actually makes it out of the vm?

Reply:
You are right, thanks, will update